packages/next-upgrade/index.ts

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env node
+
+import prompts from 'prompts'
+import fs from 'fs'
+import { execSync } from 'child_process'
+import path from 'path'
+import { compareVersions } from 'compare-versions'
+import yaml from 'yaml'
+
+type Version = 'latest' | 'rc' | 'canary'
+
+interface Response {
+  version: Version
+}
+
+async function run(): Promise<void> {
+  let version: Version
+
+  const specifiedVersion = process.argv[2]?.replace('@', '')
+  if (['latest', 'rc', 'canary'].includes(specifiedVersion)) {
+    // Shortcut
+    version = specifiedVersion as Version
+  } else {
+    const response: Response = await prompts({
+      type: 'select',
+      name: 'version',
+      message: 'What version of Next.js do you want to upgrade to?',
+      choices: [
+        {
+          title: 'Latest Stable',
+          value: 'latest',
+          description: 'Recommended for production',
+        },
+        {
+          title: 'Latest Release Candidate',
+          value: 'rc',
+          description: 'Recommended for Turbopack compatibility',
+        },
+        {
+          title: 'Canary',
+          value: 'canary',
+          description: 'Latest in-development version',
```

Comment:
@bgw @elsigh thoughts on these descriptions?

Reply:
Sent you what AI said ;)